src/pages/registration/resource_type_tab/sub_type_forms/research_data_types/RelatedResourceRow.tsx

```diff
@@ -1,53 +1,46 @@
+import { SearchListItem } from '../../../../../components/styled/Wrappers';
+import { RegistrationSearchItem } from '../../../../../types/registration.types';
 import { API_URL } from '../../../../../utils/constants';
 import { dataTestId } from '../../../../../utils/dataTestIds';
 import { useFetch } from '../../../../../utils/hooks/useFetch';
-import { getTitleString } from '../../../../../utils/registration-helpers';
-import { getRegistrationLandingPagePath } from '../../../../../utils/urlPaths';
 
-interface RelatedResourceRowRowProps {
+interface RelatedResourceRowProps {
   uri: string;
   removeRelatedResource: () => void;
 }
 
-export const RelatedResourceRow = ({ uri, removeRelatedResource }: RelatedResourceRowRowProps) => {
+export const RelatedResourceRow = ({ uri, removeRelatedResource }: RelatedResourceRowProps) => {
   const { t } = useTranslation();
   const isInternalRegistration = uri.includes(API_URL);
-  const [registration, isLoadingRegistration] = useFetch<Registration>({ url: isInternalRegistration ? uri : '' });
+  const [registration, isLoadingRegistration] = useFetch<RegistrationSearchItem>({
+    url: isInternalRegistration ? uri : '',
+  });
   const [confirmRemoveRelation, setConfirmRemoveRelation] = useState(false);
 
   return (
-    <ListItem disableGutters>
+    <>
       {isLoadingRegistration ? (
-        <Skeleton width="30%" />
+        <ListItem disablePadding>
+          <Skeleton width="30%" />
+        </ListItem>
+      ) : isInternalRegistration && registration ? (
+        <SearchListItem sx={{ borderLeftColor: 'registration.main' }} key={registration.identifier}>
+          <RegistrationListItemContent
+            registration={registration}
+            onRemoveRelated={() => setConfirmRemoveRelation(true)}
+          />
+        </SearchListItem>
       ) : (
-        <Box
-          sx={{
-            display: 'flex',
-            flexDirection: { xs: 'column', sm: 'row' },
-            alignItems: 'center',
-            gap: '0.25rem 1rem',
-          }}>
-          {isInternalRegistration ? (
-            <Link
-              data-testid={dataTestId.registrationWizard.resourceType.relatedRegistrationLink(
-                registration?.identifier ?? ''
-              )}
-              component={RouterLink}
-              to={getRegistrationLandingPagePath(registration?.identifier ?? '')}>
-              {getTitleString(registration?.entityDescription?.mainTitle)}
-            </Link>
-          ) : (
-            <Link data-testid={dataTestId.registrationWizard.resourceType.externalLink} href={uri}>
-              {uri}
-            </Link>
-          )}
+        <ListItem sx={{ display: 'flex', gap: '1rem' }}>
+          <Link data-testid={dataTestId.registrationWizard.resourceType.externalLink} href={uri}>
+            {uri}
+          </Link>
           <Button
             size="small"
             variant="outlined"
```

Comment:
Litt på siden, men `data-testid` på linje 48 er vel feil her? Den bruker `registration.identifier`, men dette er for en ekstern link, så den har jo aldri noen identifier (som også stemmer overens med sjekken på linje 32).